charts/spire/values.yaml

```diff
@@ -1,54 +1,1549 @@
-# You can enable config/features that affect all services here.
+# Default configuration for Spire chart
+# SPDX-License-Identifier: APACHE-2.0
+
+## @section Global parameters
+## Note: the paramter values specified here will override the chart level values for these parameters.
+##
 global:
   k8s:
-    # -- This is the value of your clusters `kubeadm init --service-dns-domain` flag
+    ## @param global.k8s.clusterDomain Cluster domain name configured for Spire install
+    ##
     clusterDomain: cluster.local
   spire:
-    # -- The name of the Kubernetes cluster (`kubeadm init --service-dns-domain`)
-    clusterName: example-cluster
-    # -- The trust domain to be used for the SPIFFE identifiers
-    trustDomain: example.org
-    # -- Set the jwt issuer
-    jwtIssuer: oidc-discovery.example.org
-    # -- Override all instances of bundleConfigMap
+    ## @param global.spire.bundleConfigMap A configmap containing the Spire bundle
```

Comment:
Im not familiar with the syntax youre using. The `@param` and `@section` markers. what do they do?

Reply:
Hi @faisal-memon , its just the convention used by bitnami charts and I don't think it offers any specific benefits beyond tagging the different sections. Looking through the helm reference, they are suggesting just starting the comment with the variable name - https://helm.sh/docs/chart_best_practices/values/ . I am happy to change if this format looks better. I will be fixing the linting errors (training spaces) and seeing if the docs need any updates over the next couple of days and sending an updated version.

Reply:
Thanks for working on this! :)

Currently all the doc strings are formatted for reading by https://github.com/norwoodj/helm-docs and its generating the README.md automatically. Its why the tests are broken on the PR. So the structure of the comments is important unless we change the tool being used to generate the docs. If you know a better one then the helm-docs one, that would be good to discuss. We've hit some limitations with helm-docs. Otherwise, we probably want to stick to the existing formatting structure so it works with the existing tooling.

Reply:
Lets go with this, will dig into the format and rework.